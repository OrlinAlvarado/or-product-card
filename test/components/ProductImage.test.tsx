import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';


describe('ProductImage', () => {
    test('debe mostrar el componente correctamente la imagen', () => {
        const wrapper = renderer.create(
            <ProductImage img={ product2.img } />
        );
        
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
  
    test('debe mostrar el componente dentro del componente padre', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage  />
                    )
                }
            </ProductCard>
        );
        
        expect(wrapper.toJSON()).toMatchSnapshot();
    })

});
