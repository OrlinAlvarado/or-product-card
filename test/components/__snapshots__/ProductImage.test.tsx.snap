// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente correctamente la imagen 1`] = `
<img
  alt="Coffee Mug"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage debe mostrar el componente dentro del componente padre 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
